### PR TITLE
[Role] BREAKING CHANGE: `az role assignment create`: `--scope` is now a required argument

### PR DESCRIPTION
**Pull request by jiasli**
Close https://github.com/Azure/azure-cli/issues/24753

**Related command**
`az role assignment create`

**Description**
To adhere to the Principle of Least Privilege and avoid over-scoped role assignment on the subscription, `--scope` is now a required argument.

A warning for this breaking change was announced in https://github.com/Azure/azure-cli/pull/24755.

See https://github.com/Azure/azure-cli/issues/24753 for detailed explanation.

The corresponding change on `az vm/vmss` is https://github.com/Azure/azure-cli/pull/27657.

**Testing Guide**
```sh
az role assignment create --assignee-object-id xxx --scope /subscriptions/xxx/resourceGroups/xxx --role reader

az role assignment create --assignee-object-id xxx --role reader
# This fails: the following arguments are required: --scope
```

**History Notes**
[Role] BREAKING CHANGE: `az role assignment create`: `--scope` is now a required argument.
[Role] BREAKING CHANGE: `az role assignment create`: Remove `--resource-group` argument.

### CODE REVIEW

**Comment by yonzhan** [comment]
Role

File: src/azure-cli/azure/cli/command_modules/role/_validators.py [hunk 1]
```diff
@@ -84,11 +84,13 @@ def process_assignment_namespace(cmd, namespace):  # pylint: disable=unused-argu
...
+        if getattr(namespace, arg, None) == "":
             # Get option name, like resource_group_name -> --resource-group
             option_name = cmd.arguments[arg].type.settings['options_list'][0]
             raise CLIError(non_empty_msg.format(option_name))
 
-    resource_group = namespace.resource_group_name
-    if namespace.scope and resource_group and getattr(resource_group, 'is_default', None):
-        namespace.resource_group_name = None  # drop configured defaults
+    # `az role assignment create` doesn't support resource_group_name
+    if hasattr(namespace, "resource_group_name"):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/role/_validators.py:93]: We can mention removing `--resource-group` in history notes. This is also a breaking change customers need to pay attention

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/role/_validators.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_validators.py:93]: Nice catch! Added to the **History Notes** section.

File: src/azure-cli/azure/cli/command_modules/role/_validators.py [hunk 2]
```diff
@@ -84,11 +84,13 @@ def process_assignment_namespace(cmd, namespace):  # pylint: disable=unused-argu
     non_empty_msg = "usage error: {} can't be an empty string. Either omit it or provide a non-empty string."
 
     for arg in non_empty_args:
-        if getattr(namespace, arg) == "":
+        if getattr(namespace, arg, None) == "":
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/role/_validators.py:87]: Is there any reason to add the default value for `getattr`?

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_validators.py:87]: Good question. After removing `resource_group_name` from `az role assignment create`, `resource_group_name` will not exist in `namespace`. If we don't add a default value here, `getattr` will fail for `resource_group_name` under `az role assignment create`.

**Review by bebound — APPROVED** [review]

**Pull request merged**